src/plugins/input_control_vis/public/control/control.ts

```diff
@@ -81,9 +81,10 @@ export abstract class Control<FilterManager extends BaseFilterManager> {
   abstract destroy(): void;
 
   format = (value: any) => {
+    const indexPattern = this.filterManager.getIndexPattern();
     const field = this.filterManager.getField();
-    if (field?.format?.convert) {
-      return field.format.convert(value);
+    if (field) {
+      return indexPattern.getFormatterForField(field).convert(value);
```

Comment:
since there is no more check for the existence of the convert function, can we. be sure that a formatter exists, or a default one is returned in case it doesn't?

Reply:
`indexPattern.getFormatterForField` calls `fieldFormats.getDefaultInstance` which calls `FieldFormatsRegistry.getDefaultInstance` which calls `FieldFormatsRegistry.getDefaultInstancePlain` which calls `FieldFormatsRegistry.getInstance` which throws if it doesn't find a formatter class. 

As best I can tell, we can be certain that there will be a formatter or the code will throw. I'd appreciate verification.

Reply:
verified, I do agree.